scripts/build.py

```diff
@@ -94,51 +105,62 @@ def _minify(source_path, target_path):
     subprocess.check_call(cmd, shell=True)
 
 
-def _join_files(source_paths, target_file_path):
+def write_to_file_stream(file_stream, content):
+    """Write to a file object using provided content.
+
+    Args:
+        file_stream: file. A stream handling object to do write operation on.
+        content: str. String content to write to file object.
+    """
+    file_stream.write(content)
+
+
+def _join_files(source_paths, target_file_stream):
     """Writes multiple files into one file.
 
     Args:
         source_paths: list(str). Paths to files to join together.
-        target_file_path: str. Path to location of the joined file.
+        target_file_stream: file. A stream object of target file.
     """
-    ensure_directory_exists(target_file_path)
-    with open(target_file_path, 'w+') as target_file:
-        for source_path in source_paths:
-            with open(source_path, 'r') as source_file:
-                target_file.write(source_file.read())
+    for source_path in source_paths:
+        with open(source_path, 'r') as source_file:
+            write_to_file_stream(target_file_stream, source_file.read())
 
 
-def _minify_and_create_sourcemap(source_paths, target_file_path):
-    """Minifies multiple files into one file and generates source map
-    for that file.
+def _minify_and_create_sourcemap(source_path, target_file_path):
+    """Minifies and generates source map for a JS file.
 
     Args:
-        source_paths: list(str). Paths to files to join and minify.
-        target_file_path: str. Path to location of the joined file.
+        source_path: str. Path to JS file to minify.
+        target_file_path: str. Path to location of the minified file.
     """
-
+    print 'Minifying and creating sourcemap for %s' % source_path
     ensure_directory_exists(target_file_path)
```

Comment:
Oh it was meant to create the directory tree (parent folder) for the minified third_party_min.js. Doesn't look like its necessary though because `build_third_party_libs()` already create the directory. I removed the function call.